crypto/mem.c

```diff
@@ -165,13 +166,25 @@ void ossl_malloc_setup_failures(void)
 {
     const char *cp = getenv("OPENSSL_MALLOC_FAILURES");
 
-    if (cp != NULL && (md_failstring = strdup(cp)) != NULL)
+    if (cp != NULL) {
+        md_fs_orig = OPENSSL_strdup(cp);
+        md_failstring = md_fs_orig;
         parseit();
+    }
     if ((cp = getenv("OPENSSL_MALLOC_FD")) != NULL)
         md_tracefd = atoi(cp);
     if ((cp = getenv("OPENSSL_MALLOC_SEED")) != NULL)
         srandom(atoi(cp));
 }
+
+void ossl_free_mdebug_internals(void)
+{
+    if (md_fs_orig != NULL) {
+        free(md_fs_orig);
+        md_fs_orig = NULL;
```

Comment:
Since `md_failstring` points into `md_fs_orig`, that should be set to `NULL` here too